Update configuration for new hosts

super-cooper opened the pull request:
- Add DB_TAG to env
- Use new, merged Let's Encrypt certs for Reference API
- Update docker-compose for new portal hosts
- Misc fixes

These changes are already in use on the new portal hosts, but this is a formal request to merge them into camino

super-cooper committed:
Remove redis from dev compose

super-cooper committed:
Add database service to dev compose

super-cooper committed:
Add DB_TAG to dev env

super-cooper committed:
Update dev compose configuration

super-cooper committed:
Misc fixes

super-cooper committed:
Use correct hostname for dev referenceapi

super-cooper committed:
Add MariaDB tag to prod environment

super-cooper committed:
Use new, merged Let's Encrypt certs for Reference API

super-cooper committed:
Update prod docker-compose for new portal host

msherman64 commented on conf/nginx/conf.d/referenceapi.server.conf:
```diff
@@ -1,14 +1,14 @@
   listen      443 ssl;
 
-  server_name    api.chameleoncloud.org api.tacc.chameleoncloud.org;
+  server_name    api.dev.chameleoncloud.org;
```
why is `dev` hardcoded here?

super-cooper replied:
You're right, this was left in by accident... Unfortunately, I'm not sure of the best way to switch on this value in nginx configuration. Trying to figure out a nice way to do this.

super-cooper committed:
Revert "Remove redis from dev compose"

This reverts commit 54530a5c6af2f35e6c998ff306ad73ddee90d6d7.

super-cooper committed:
Return redis to prod compose

super-cooper committed:
Re-add redis dependency to Celery

super-cooper committed:
Nginx switches on hostname

Nginx can dynamically decide what URL the reference API should listen on based on the container's hostname. This is the simplest solution since Nginx is not able to read environment variables.

msherman64 reviewed (approved):
LGTM, thanks!
